File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/aws/s3/MetaHeaderTest.java
package uk.nhs.nhsx.core.aws.s3;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.MatcherAssert.assertThat;

public class MetaHeaderTest {

    @Test
    public void s3headerName() throws Exception {
        assertThat(new MetaHeader("key", "value").asS3MetaName(), equalTo("key"));
    }

    @Test
    public void httpHeaderName() throws Exception {
        assertThat(new MetaHeader("key", "value").asHttpHeaderName(), equalTo("X-Amz-Meta-key"));
    }
}

File: src/aws/lambdas/incremental_distribution/src/test/java/uk/nhs/nhsx/core/aws/s3/MetaHeaderTest.kt
package uk.nhs.nhsx.core.aws.s3

import org.hamcrest.CoreMatchers.equalTo
import org.hamcrest.MatcherAssert.assertThat
import org.junit.jupiter.api.Test

class MetaHeaderTest {

    @Test
    fun s3headerName() {
        assertThat(MetaHeader("key", "value").asS3MetaName(), equalTo("key"))
    }


    @Test
    fun httpHeaderName() {
        assertThat(MetaHeader("key", "value").asHttpHeaderName(), equalTo("X-Amz-Meta-key"))
    }
}